File: frontend/src/api/movies.ts
import { apiClient } from './client'

export interface MovieDto {
  id: number
  title: string
  release_year: number | null
  runtime: number | null
  vote_average: number | null
  vote_count: number | null
  bayesian_rating: number | null
  overview: string | null
  poster_url: string | null
  genres: string[]
  director: string | null
  top_cast: string[]
  original_language: string | null
  trailer_youtube_key: string | null
  umap_x: number | null
  umap_y: number | null
}

/**
 * Fetch full metadata for a single movie.
 *
 * @param movieId - TMDB integer movie ID.
 * @returns MovieDto on success.
 */
export async function getMovieFetcher(movieId: number): Promise<MovieDto> {
  return apiClient<MovieDto>(`/movies/${movieId}`)
}

/**
 * Fetch full metadata for a batch of movies in a single request.
 * Unknown IDs are silently omitted from the response.
 *
 * @param ids - Up to 200 TMDB integer movie IDs.
 * @returns Array of MovieDto in the same order as ids (missing IDs dropped).
 */
export async function getMoviesBatchFetcher(ids: number[]): Promise<MovieDto[]> {
  if (ids.length === 0) return []
  return apiClient<MovieDto[]>('/movies/batch', {
    method: 'POST',
    body: JSON.stringify({ ids }),
  })
}

File: mcp_server/clients/movies.py
from mcp_server.clients.http import BaseClient


class MoviesClient(BaseClient):
    """HTTP client for the /movies domain.

    All methods map 1-to-1 to backend routes under the ``/movies`` prefix.
    Non-2xx responses are raised as ``BackendError``.
    """

    async def get_movie(self, movie_id: int) -> dict:
        """GET /movies/get/{id} — full metadata for one movie by TMDB ID.

        Args:
            movie_id: Integer TMDB movie ID.

        Returns:
            ``MovieDto`` dict with title, overview, genres, release_year, and more.
        """
        resp = await self._http.get(f"/movies/get/{movie_id}")
        self._check(resp)
        return resp.json()

    async def get_movies_batch(self, movie_ids: list[int]) -> list:
        """POST /movies/get_batch — full metadata for up to 200 movies in one call.

        Args:
            movie_ids: List of up to 200 integer TMDB movie IDs.

        Returns:
            List of ``MovieDto`` dicts.
        """
        resp = await self._http.post("/movies/get_batch", json={"ids": movie_ids})
        self._check(resp)
        return resp.json()
